lib/datadog/core/environment/execution.rb

```diff
@@ -10,7 +10,22 @@ class << self
           # This can be used to make decisions about when to enable
           # background systems like worker threads or telemetry.
           def development?
-            !!(repl? || test? || rails_development?)
+            !!(repl? || test? || rails_development? || webmock_enabled?)
+          end
+
+          # WebMock stores the reference to `Net::HTTP` with constant `OriginalNetHTTP`, and when WebMock enables,
+          # the adapter swaps `Net::HTTP` reference to its mock object, @webMockNetHTTP.
+          #
+          # Hence, we can detect by
+          #   1. Checking if `Net::HTTP` is referring to mock object
+          #   => ::Net::HTTP.equal?(::WebMock::HttpLibAdapters::NetHttpAdapter.instance_variable_get(:@webMockNetHTTP))
+          #
+          #   2. Checking if `Net::HTTP` is referring to the original one
+          #   => ::Net::HTTP.equal?(::WebMock::HttpLibAdapters::NetHttpAdapter::OriginalNetHTTP)
+          def webmock_enabled?
+            defined?(::WebMock) &&
```

Comment:
We check for `defined?(::WebMock)`, but then access `::WebMock::HttpLibAdapters::NetHttpAdapter`, which I think can be dangerous. I suggest:

```suggestion
            defined?(::WebMock::HttpLibAdapters::NetHttpAdapter) &&
```